Formula/o/opencilk.rb

```diff
@@ -0,0 +1,732 @@
+class Opencilk < Formula
+  desc "Task-parallel programming platform"
+  homepage "https://www.opencilk.org/"
+  url "https://github.com/OpenCilk/opencilk-project/archive/refs/tags/opencilk/v3.0.tar.gz"
+  version "19.1.7"
+  sha256 "0e23756c05c97596009f6a2d8d1625ed34fe556a44067413089d4ca321a0c4d7"
+  # The OpenCilk Project is under the MIT License and the Apache License v2.0 with LLVM Exceptions
+  license all_of: [
+    "MIT",
+    "Apache-2.0" => { with: "LLVM-exception" },
+  ]
+
+  # This formula is based heavily on the llvm@19 formula.
+
+  # Clang cannot find system headers if Xcode CLT is not installed
+  pour_bottle? only_if: :clt_installed
+
+  keg_only :shadowed_by_macos, "macOS provides clang"
```

Comment:
Does opencilk provide clang?

Reply:
Yes, OpenCilk provides a modified version of clang.

The modified clang should support everything that ordinary clang supports plus some additional features.

Reply:
In that case it should also declare a conflict with all llvm formulae.

Reply:
I can do that, although the audit now complains that the llvm formulae should also have a conflict declared with opencilk.  Do I need to update those formulae as well in this PR?

Reply:
Alternatively, I could rename the clang binaries that opencilk provides, to make them distinct from those provided by the llvm formulae.  I believe there are other binaries or libraries in the LLVM toolchain that may conflict between opencilk and llvm.  But I suspect those binaries and libraries would be rarely used directly.  In addition, the opencilk and llvm formulae are keg-only already, which should further mitigate risks.  I figured I should offer to change opencilk's clang-binary names, in case it would help make the opencilk formula acceptable, at least in the short term.